membership-attribute-service/app/services/AttributesFromZuora.scala

```diff
@@ -60,40 +60,41 @@ object AttributesFromZuora extends LoggingWithLogstashFields {
       val (fromWhere: String, attributes: Option[Attributes]) = attributesFromSomewhere
 
       if(fromWhere == "Zuora") {
-        attributesFromDynamo flatMap { (dynamoAttributes: Option[Attributes]) =>
-          val shouldSkipUpdate: Boolean = dynamoAndZuoraAgree(dynamoAttributes, attributes, identityId)
-          val zuoraAttributesWithAdfree: Option[Attributes] = attributesWithFlagFromDynamo(attributes, dynamoAttributes)
-          attributesAfterDynamoUpdate(identityId, shouldSkipUpdate, zuoraAttributesWithAdfree, dynamoAttributeService) map { (fromWhere, _) }
+        attributesFromDynamo map { (dynamoAttributes: Option[Attributes]) =>
+          val canSkipCacheUpdate: Boolean = dynamoAndZuoraAgree(dynamoAttributes, attributes, identityId)
+          val zuoraAttributesWithAdfree: Option[Attributes] = attributesWithAdFreeFlagFromDynamo(attributes, dynamoAttributes)
+
+          if(!canSkipCacheUpdate) {
+            updateCache(identityId, zuoraAttributesWithAdfree, dynamoAttributeService).onFailure {
+              case e: Throwable =>
+                log.error(s"Future failed when attempting to update cache. Update was needed? $canSkipCacheUpdate", e)
```

Comment:
I know you haven't introduced this here, but do we need two separate log lines for this?

Reply:
I wanted to be able to see the attributes from Zuora in Kibana in case it's needed for debugging, but an error log to get picked up by Sentry if updates are failing.